csrgraph/graph.py

```diff
@@ -518,33 +523,66 @@ def read_edgelist(f, directed=True, sep=r"\s+", header=None, keep_default_na=Fal
     # Get the input data type
     if nnodes > UINT32_MAX:
         dtype = np.uint64
-    else:
+    elif nnodes > UINT16_MAX:
         dtype = np.uint32
+    else:
+        dtype = np.uint16
+    print(f"nnodes: {nnodes} dtype: {dtype}")
     name_dict = dict(zip(names,
                          np.arange(names.shape[0], dtype=dtype)))
+    print('memory1', memory_profiler.memory_usage()[0])
+    print('elist size 1', sys.getsizeof(elist))
     elist.src = elist.src.map(name_dict)
     elist.dst = elist.dst.map(name_dict)
+    # convert names from float to uint16
+    if dtype == np.uint16:
+        elist.src = np.uint16(elist.src.values)
+        elist.dst = np.uint16(elist.dst.values)
+    print('memory2', memory_profiler.memory_usage()[0])
+    print('elist size 2', sys.getsizeof(elist))
+    # convert weights to float32 (NEED TO DOUCLBE CHECK this doesn't affect the embedding output)
+    elist.weight = np.float16(elist.weight.values)
```

Comment:
`float16` is more compact in memory, but it's a storage format first, so it can actually end up being drastically slower than `float32` on many types of operations.

I generally recommend against using it.